assets/js/components/TourTooltip.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * TourTooltip component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import Card, { CardActions } from '@material/react-card';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import '@material/react-card/dist/card.css';
+
+/**
+ * Internal dependencies
+ */
+import CloseIcon from '../../svg/close.svg';
+import Button from './Button';
+
+/**
+ * Creates an array of length equal to size with incremental values starting at 0.
+ *
+ * @since n.e.x.t
+ *
+ * @param {number} size Array length.
+ * @return {number[]} Array of incremental numbers.
+ * @example createIncrementalArrayBySize(3); // [0, 1, 2]
+ */
+const createIncrementalArrayBySize = ( size ) => new Array( size ).fill().map( ( _, i ) => i );
```

Comment:
Looks like this function is used only once in the TourTooltip component. Why not use it directly there?

```javascript
const indicatorArray = new Array( size ).fill().map( ( _, i ) => i );
const getIndicatorClassName = ( indicatorIndex ) => ...
```

Reply:
@eugene-manuilov Forgive the wordy response! I tried my best to refine it down to be as short as possible and still convey the intended info. 😅

I created this function to prevent technical debt by adding clarity around an otherwise confusing or scary-looking piece of logic via explicit function naming. I know the `Array.fill.map` trick now but cannot guarantee I (or someone else) will know it in the future. This is useful while debugging or otherwise trying to understand component logic — when working memory is a significant factor in how long it takes to fix an issue. I prefer this method over comments alone as comments are easy to ignore when skimming through logic and have a greater chance to contain stale information.

This has the added benefit of providing example usage (and data return) on-hover, in-editor, which can help if there's any confusion with function naming (JSDoc description can go even further). While not a big deal in this case as there's nothing to be mentally parsed above `indicatorArray` — this abstraction keeps the render function clean (easy to parse quickly) by keeping related logic colocated. There are no comments or unnecessary logic visually breaking the top-down flow of logic.

This also decouples the logic within `createIncrementalArrayBySize` away from the render method of the component. As a result, we can assume the function does not contain any side effects that are component related (props, state, anything component-specific) — meaning we can ignore it when mentally parsing through component logic as any unit complex enough to warrant such a worry should be guarded by unit tests in the first place. Logic & implementation details of the function can be ignored altogether — only the data returned is of concern. 

Significantly less of a concern to me — and somewhat counterintuitively, this can result in micro-perf benefits under certain circumstances due to JIT compiler optimizations (otherwise, within the margin of error). This is usually only relevant when being called within a loop, but it doesn't hurt to mention.

Because I'm a caveman and probably explained things poorly, I'll end this with a Martin quote: *“Indeed, the ratio of time spent reading versus writing is well over 10 to 1. We are constantly reading old code as part of the effort to write new code. ...[Therefore,] making it easy to read makes it easier to write.”*

Let me know how you feel about my thoughts above! In no way am I particularly attached to them. I'm happy to make the change if it seems more confusing than helpful. Just wanted to shed light on the thought process behind the decision. 🙏

Reply:
Well, I have nothing against it, but in this case, let's move it to util functions then. So, we can use it in different components if need be.